packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts

```diff
@@ -0,0 +1,31 @@
+import request from 'request';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import config from 'temp/config';
+import { GraphQLSitemapService } from '@sitecore-jss/sitecore-jss/site';
+
+const ABSOLUTE_URL_REGEXP = '^(?:[a-z]+:)?//';
+
+const sitemapApi = async (req: NextApiRequest, res: NextApiResponse): Promise<NextApiResponse | void> => {
+  const { query: { id } } = req;
+  // create sitemap graphql service
+  const sitemapService = new GraphQLSitemapService({
+    endpoint: config.graphQLEndpoint,
+    apiKey: config.sitecoreApiKey,
+    siteName: config.jssAppName,
+  });
+
+  const existSitemap = await sitemapService.getSitemap(id as string);
+
+  if (existSitemap) {
+    const isAbsoluteUrl = existSitemap.match(ABSOLUTE_URL_REGEXP);
+    const sitemapUrl = isAbsoluteUrl ? existSitemap : `${config.sitecoreApiHost}${existSitemap}`;
+    res.setHeader('Content-Encoding', 'gzip');
```

Comment:
Actually Next.js might [gzip by default](https://nextjs.org/blog/next-9-0-7#gzip-compression-by-default), in which case this header should already be there (you'll need to check). If it doesn't (possible for API routes since these will be serverless functions), we would need to gzip ourselves. 

Could use Node 'zlib'.
```
import { createGzip } from 'zlib';
...
return request(sitemapUrl).pipe(createGzip()).pipe(res);
```

Reply:
Yes, the Nextjs use gzip by default. I removed content-encoding